ocaml-lsp-server/test/e2e-new/polarity_search.ml

```diff
@@ -0,0 +1,320 @@
+open Test.Import
+module Req = Ocaml_lsp_server.Custom_request.PolaritySearch
+
+module Util = struct
+  let call_search position query client =
+    let uri = DocumentUri.of_path "test.ml" in
+    let text_document = TextDocumentIdentifier.create ~uri in
+    let params =
+      Req.Request_params.create text_document position query
+      |> Req.Request_params.yojson_of_t
+      |> Jsonrpc.Structured.t_of_yojson
+      |> Option.some
+    in
+    let req =
+      Lsp.Client_request.UnknownRequest { meth = "ocamllsp/polaritySearch"; params }
+    in
+    Client.request client req
+  ;;
+
+  let test ~line ~character ~query source =
+    let position = Position.create ~character ~line in
+    let request client =
+      let open Fiber.O in
+      let+ response = call_search position query client in
+      Test.print_result response
+    in
+    Helpers.test source request
+  ;;
+end
+
+let%expect_test "Search for a simple query that takes an int and returns a string" =
+  let source = "" in
+  let line = 1 in
+  let character = 0 in
+  Util.test ~line ~character ~query:"-int +string" source;
+  [%expect
+    {|
+    [
+      {
+        "name": "string_of_int",
```

Comment:
Unfortunately no, but we can probably filter the result of test just keeping value prefixed by a given prefix? WDYT @PizieDust ?

Reply:
That's definitely an improvement, but I'm kind of surprised that we can't write our fixtures. Is it because the test doesn't allow us to setup a dune project that doesn't use the stdlib?

Reply:
Ah yes, we can probably make a test without any dependencies (and that pass the flag `nostdlib`